tests/transforms/test_qcut.py

```diff
@@ -2917,3 +2918,134 @@ def test_validate_fragment_sizes(self, max_wires_by_fragment, max_gates_by_fragm
                 max_wires_by_fragment=max_wires_by_fragment,
                 max_gates_by_fragment=max_gates_by_fragment,
             )
+
+
+def make_weakly_connected_tape(
+    fragment_wire_sizes=[3, 5],
+    single_gates_per_wire=1,
+    double_gates_multiplier=1.5,
+    inter_fragment_gate_wires={(0, 1): 1},
+    repeats=2,
+):
+    """Helper function for making random weakly connected tapes."""
```

Comment:
Yes I'm a bit hesitating to make it random too, But the randomness here is actually helping me discover bugs. For completely disjoint fragments, no matter how random it is, there should not be any cut made. For weakly connected fragments, the total number of cuts should not exceed twice the number of gates connecting fragments, which is not a tight bound for small graphs. If there's something failing here, it means my understanding is off.

But for CI tests I know it would be an inconvenience for others if the test failed randomly, maybe I can fix the seed inside the CI to make sure things pass deterministically. But for local test, it can be random. Thoughts? @trbromley

Reply:
I'm pretty convinced by your reasoning here! Only thing that might stop this is if it goes against PL convention.

Reply:
Now that that [unrelated test](https://github.com/PennyLaneAI/pennylane/blob/4df8dc55521eb0e60e10def3a3ad374020eba189/tests/test_optimize_shot_adapative.py#L132) is passing after I switched to vanilla numpy's `RandomState`. Could it be that making calls to `pennylane.numpy` 's `random` module affects things elsewhere?